openedx/features/cms/helpers.py

```diff
@@ -62,3 +67,62 @@ def set_rerun_course_module_dates(source_course, re_run, user):
             meta_data['due'] = re_run_start_date - due_date_delta
 
         _save_xblock(user, re_run_xblock, metadata=meta_data)
+
+
+def set_ora_dates(re_run_subsections, source_course_start_date, re_run_start_date, user):
+    """
+    This method is responsible for updating all dates in ORA i.e submission, start, due etc, for
+    the re-run according to source course. This is achieved by calculating new dates for ORA based
+    on delta value.
+    :param re_run_subsections: list of subsection in a (re-run) course
+    :param source_course_start_date: course start date of source course
+    :param re_run_start_date: course start date of re-run course
+    :param user: user that created this course
+    """
+    def update_date_by_delta(date_to_update):
```

Comment:
let's add a default flag in this, and move the 2 if(s) inside this and also update the value from inside this block and return boolean regarding if updation was required or not.
for the return value do something like this:
```
to_update = False
to_update = update_date_by_delta(ora.submission_due, DEFAULT_START) or to_update
# this could also be done as 
# to_update += update_date_by_delta(ora.submission_due, DEFAULT_START)
```

Reply:
I guess we cannot follow this approach here, we also have to update `ora.submission_due`. Though I have updated this code in another way, please have a look.

Relevant link to our earlier discussion, [Is Python pass-by-reference or pass-by-value](https://robertheaton.com/2014/02/09/pythons-pass-by-object-reference-as-explained-by-philip-k-dick/)